code/modules/recycling/sortingmachinery.dm

```diff
@@ -48,37 +42,48 @@
 		if(WP.use(3))
 			user.visible_message("[user] wraps the package in festive paper!")
 			giftwrapped = 1
-			if(istype(wrapped, /obj/structure/closet/crate))
-				icon_state = "giftcrate"
-			else
-				icon_state = "giftcloset"
-			if(WP.amount <= 0 && !WP.loc) //if we used our last wrapping paper, drop a cardboard tube
-				new /obj/item/weapon/c_tube( get_turf(user) )
+			icon_state = "gift[icon_state]"
 		else
 			user << "<span class='warning'>You need more paper!</span>"
 
+/obj/structure/bigDelivery/relay_container_resist(mob/living/user, obj/O)
+	if(istype(loc, /atom/movable))
+		var/atom/movable/AM = loc //can't unwrap the wrapped container if it's inside something.
+		AM.relay_container_resist(user, O)
+		return
+	user << "<span class='notice'>You lean on the back of [O] and start pushing to rip the wrapping around it.</span>"
+	if(do_after(user, 50, target = O))

```

Comment:
only takes 5 seconds to rip the paper. I didn't want to punish people in a locked and wrapped closet too much.